doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+  solved in the Real `domain` then only `\{0\}` is returned.
+
+
+Why do we use Sets as an output type?
+-------------------------------------
+
+SymPy has a well developed sets module, which can represent most of the set
+containers in Mathematics such as:
+
+ * ``FiniteSet``
+   Represents a finite set of discrete numbers.
+
+ * ``Interval`` 
+   Represents a real interval as a Set.
+
+ * ``ProductSet``
+   Represents a Cartesian Product of Sets.
+
+ * ``ImageSet``
+   Represents the Image of a set under a mathematical function
+
+    >>> from sympy import ImageSet, S, Lambda
+    >>> from sympy.abc import x
+    >>> from sympy.sets import ImageSet
+    >>> squares = ImageSet(Lambda(x, x**2), S.Naturals)  # {x**2 for x in N}
+    >>> 4 in squares
+    True
+
+ * ``ComplexPlane``
+   Represents the Set of all Complex Numbers
+
+ * ``ConditionSet``
+   Represents the Set of elements which satisfies a given condition
+
+Also, the predefined set classes such as:
+
+ * ``Naturals`` `\mathbb{N}`
+   Represents the natural numbers (or counting numbers) which are all
+   positive integers starting from 1 
+
+ * ``Naturals0`` `\mathbb{W}`
+   Represents the whole numbers which are all the non-negative integers,
+   inclusive of zero
+
+ * ``Integers`` `\mathbb{Z}`
+   Represents all integers: positive, negative and zero. This set is also
+   available as the Singleton, ``S.Integers``.
+
+ * ``Reals`` `\mathbb{R}`
+   Represents the Set of all Real numbers.
+
+ * ``Complexes`` `\mathbb{C}`
+   Represents the Set of all Complex numbers.
+
+It is capable of most of the set operations in Mathematics:
+
+ * ``Union``
+ * ``Intersection``
+ * ``Complement``
+ * ``SymmetricDifference``
+
+The main reason for using sets as output to solvers is that it can consistently
+represent many types of solutions. For single variable case it can represent:
+
+ * No solution (by null set)
+
+ * Finitely many solutions (by FiniteSet)
+
+ * Infinitely many solutions, both countably and uncountably infinite solutions
+   (using the ImageSet module) Interval
+
+ * There can also be bizarre solutions to equations like set of all rational number.
+
+No other programmer's object (list, dictionary, python sets) provides the
+flexibility of mathematical sets which our sets module try to emulate. The
+second reason to use sets is that they are close to the entities which
+mathematician's deals with and it makes it easier to reason about them.
+Another advantage of using objects closer to mathematical entities is that the
+user won't have to "learn" our representation and she can have her expectations
+transferred from her mathematical experience.
+
+For multivariate case we are representing solutions as a set of points in a
+n-dimensional space and a point is represented by a FiniteSet of ordered tuple.
+Please Note that, the General FiniteSet is unordered, but a FiniteSet with
+a tuple as it's first argument is ordered. For example:
```

Comment:
It could be:
``FiniteSet`` with a tuple as it's **only** argument is ordered, Since a tuple is ordered.

Because:

For example: `{1, 2}` is equivalent to `{2, 1}` : (So Unordered), but
`{(1, 2)}` is **not** equivalent to `{(2, 1)}` : So Ordered.